src/rfactor/rain.py

```diff
@@ -0,0 +1,336 @@
+
+    Translated the input file_path to the default input data used this package. This
+    functions can be used for users an example to format functions. The file is a
+    tab delimited files (extension: ``.txt``), and holds the timeseries for one
+    location. The name of the file is the tag that will be used.
+
+    Parameters
+    ----------
+    file_path: pathlib.Path
+        File path (tab delimited, .txt-extension). Headerless
+
+        - ``%d-%m-%Y %H:%M:%S``-format
+        - float
+
+    interpolate: str, default None
+        Interpolation method to use for NaN-Values. Possible values:
+        see pandas.DataFrame.interpolate.
+
+    interval: int, default np.inf
+        The max interval length over which NaN values are interpolated. The value
+        needs to fit the index of the timeseries. For example, a timeseries with
+        resolution of 10 min will have a maximum interval length of 6 hours if the
+        interval value is set to 36 (36 * 10 min = 6 hours).
+
+    threshold_outliers: int, default None
+        Set rainfall values above this threshold to NaN.
+
+
+    Returns
+    -------
+    rain : pandas.DataFrame
+        DataFrame with rainfall time series. Contains the following columns:
+
+        - *datetime* (pandas.Timestamp): Time stamp.
+        - *minutes_since* (float): Minutes since start of year.
+        - *station* (str): station identifier.
+        - *rain_mm* (float): Rain in mm.
+
+    Example
+    -------
+    Example of a rainfall file:
+
+    ::
+        2024-01-01 00:00:00	0.0
+        2024-01-01 00:10:00	0.0
+        2024-01-01 00:20:00	0.0
+        2024-01-01 00:30:00	10.5
+        2024-01-01 00:40:00	5.2
+        2024-01-01 00:50:00	1
+        2024-01-01 01:00:00	0.02
+        2024-01-01 01:10:00
+
+    """
+    df = pd.read_csv(
+        file_path,
+        sep="\t",
+        header=None,
+        names=["datetime", "rain_mm"],
+        na_values=["---", ""],
+        parse_dates=[0],
+    )
+
+    if not {"datetime", "rain_mm"}.issubset(df.columns):
+        msg = (
+            f"File '{file_path}' should should contain columns 'datetime' and "
+            f"'Value [millimeter]'."
+        )
+        raise KeyError(msg)
+
+    station, year = _extract_metadata_from_file_path(file_path)
+    df["station"] = station
+
+    # Sanitize rain outliers (negative values and values beyond threshold)
+    df.loc[df["rain_mm"] < 0, "rain_mm"] = np.nan
+    if threshold_outliers:
+        df.loc[df["rain_mm"] > threshold_outliers, "rain_mm"] = np.nan
+
+    # Short-period Nan values are interpolated
+    if interpolate:
+        # Remove consecutive NaN-values that are longer than the interval
+        is_nan = df["rain_mm"].isna()
+        group = (is_nan != is_nan.shift()).cumsum()
+        df_temp = df[is_nan].groupby(group[is_nan]).transform("size") > interval
+        indices_to_remove = df_temp[df_temp].index
+        df = df.drop(index=indices_to_remove)
```

Comment:
I opted here to remove them instead of using null values in between. Note that this could actually influence the result of more advanced interpolation methods. Please provide input on the preference.